common/lib/api-client/request-headers.js

```diff
@@ -3,13 +3,21 @@ const uuid = require('uuid')
 const {
   API: { VERSION },
   APP_VERSION,
+  CUSTOM_HEADERS,
 } = require('../../../config')
 
-module.exports = (format = 'application/vnd.api+json') => {
-  return {
+module.exports = (req, format = 'application/vnd.api+json') => {
+  const headers = {
     'User-Agent': `hmpps-book-secure-move-frontend/${APP_VERSION}`,
     Accept: `${format}; version=${VERSION}`,
     'Accept-Encoding': 'gzip',
     'Idempotency-Key': uuid.v4(),
+    ...CUSTOM_HEADERS,
   }
+
+  if (req && req.user) {
+    headers['X-Current-User'] = req.user.username
```

Comment:
I was wondering if we want to store the person's fullname instead/as well?

Reply:
It can't hurt sending it if we want, but Martin B was quite keen for us to not store people's names to leave that to HMPPS Auth. The current usecase is happy with just the username being shown, and we can extract more info from the Auth service if we need it.

There's nowhere to store it in the backend now as well :D

Reply:
Ideally we'd just send the user credentials and let the BE do what it wants with them...  validate the request...  extract the user details as required...

But in the absence of that, shouldn't we just send all the user details and let the BE choose what to use?

Reply:
I reckon this will evolve over time - currently the BE only knows about username as a first pass, but yes, in the future we will probably send the token and do all sorts of nice things.